test/e2e/runner/evidence.go

```diff
@@ -209,14 +209,25 @@ func generateDuplicateVoteEvidence(
 	if err != nil {
 		return nil, err
 	}
-	ev := types.NewDuplicateVoteEvidence(voteA, voteB, time, vals)
-	if ev == nil {
-		return nil, fmt.Errorf("could not generate evidence a=%v b=%v vals=%v", voteA, voteB, vals)
+	ev, err := types.NewDuplicateVoteEvidence(voteA, voteB, time, vals)
+	if err != nil {
+		return nil, fmt.Errorf("could not generate evidence: %w", err)
 	}
 
 	return ev, nil
 }
 
+func getRandomValidatorIndex(privVals []types.MockPV, vals *types.ValidatorSet) (types.MockPV, int32, error) {
```

Comment:
Please document this, particularly the naked integer in the return tuple. I thought at first it was just an offset into an array, but it appears to be something more.